var/spack/repos/builtin/packages/plasma/package.py

```diff
@@ -58,36 +61,63 @@ class Plasma(MakefilePackage):
     conflicts("%xl_r")
 
     patch("remove_absolute_mkl_include.patch", when="@17.1")
-    patch("add_netlib_lapacke_detection.patch", when="@17.1")
-
-    def getblaslapacklibs(self):
-        if '^netlib-lapack' in self.spec:
-            bl_attr = ':c,fortran'
-        else:
-            bl_attr = ''
-        return self.spec['lapack' + bl_attr].libs + \
-            self.spec['blas' + bl_attr].libs + \
-            find_system_libraries(['libm'])
 
+    @when("@18.9.0:")
+    def cmake_args(self):
+        if self.spec.satisfies('@:17.1'):
+            open("CMakeLists.txt", "w").write("cmake_minimum_required( VERSION 3.0 FATAL_ERROR )\n")
+
+        options = list()
+
+        options.extend([
+            "-DCMAKE_INSTALL_PREFIX=%s" % prefix,
+            "-DCMAKE_INSTALL_NAME_DIR:PATH=%s/lib" % prefix,
+            "-DBLAS_LIBRARIES=%s" % self.spec["blas"].libs.joined(";"),
+            "-DLAPACK_LIBRARIES=%s" % self.spec["lapack"].libs.joined(";")
+        ])
+
+        options += [
+            "-DBUILD_SHARED_LIBS=%s" %
+            ('ON' if ('+shared' in self.spec) else 'OFF')
+        ]
+
+        return options
+
+    # Before 18.9.0 it was an Makefile package
+    @when("@:17.1")
+    def cmake(self, spec, prefix):
+        pass
+
+    # Before 18.9.0 it was an Makefile package
+    @when("@:17.1")
+    def build(self, spec, prefix):
+        pass
+
+    # Before 18.9.0 it was an Makefile package
+    @when("@:17.1")
+    def install(self, spec, prefix):
+        self.edit(spec, prefix)
+        make()
+        make("install")
+
+    @when("@:17.1")
     def edit(self, spec, prefix):
```

Comment:
I'm not sure if this will work. It is a CmakePackage and I don't think it executes an `edit`-phase. I implemented everything needed to be done in the "old" build flow under a single (decorated) `install` method

Reply:
`edit()` method is called from install. I tried it with all combinations of versions and dependencies and it `install()`/`edit()` seemed to have done its job.